### PR TITLE
update containerservice sdk to 2022-03-02-preview

### PR DESCRIPTION
**Pull request by ms-henglu**

### CODE REVIEW

File: internal/services/containers/kubernetes_cluster_node_pool_resource.go [hunk 1]
```diff
@@ -843,7 +843,7 @@ func resourceKubernetesClusterNodePoolDelete(d *pluginsdk.ResourceData, meta int
 		return err
 	}
 
-	future, err := client.Delete(ctx, id.ResourceGroup, id.ManagedClusterName, id.AgentPoolName)
+	future, err := client.Delete(ctx, id.ResourceGroup, id.ManagedClusterName, id.AgentPoolName, nil)
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_node_pool_resource.go:846]: We should be defaulting this toggle to `true`

File: internal/services/containers/kubernetes_cluster_resource.go [hunk 2]
```diff
@@ -1908,7 +1908,7 @@ func resourceKubernetesClusterDelete(d *pluginsdk.ResourceData, meta interface{}
 		}
 	}
 
-	future, err := client.Delete(ctx, id.ResourceGroup, id.ManagedClusterName)
+	future, err := client.Delete(ctx, id.ResourceGroup, id.ManagedClusterName, nil)
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_resource.go:1911]: Same here

**Review by stephybun — CHANGES_REQUESTED** [review]

**Comment by ms-henglu** [comment]
Hi @stephybun , I've changed them to true, would you please take another look? Thanks!

**Review by stephybun — COMMENTED** [review]
There seem to be new test failures as a result of the upgrade that are not happening on main
```
------- Stdout: -------
=== RUN   TestAccKubernetesCluster_addonProfileIngressApplicationGateway_subnetCIDR
=== PAUSE TestAccKubernetesCluster_addonProfileIngressApplicationGateway_subnetCIDR
=== CONT  TestAccKubernetesCluster_addonProfileIngressApplicationGateway_subnetCIDR
testcase.go:110: Step 1/4 error: Error running apply: exit status 1
Error: creating Cluster: (Managed Cluster Name "acctestaks220509082729837134" / Resource Group "acctestRG-aks-220509082729837134"): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="IngressAppGwAddonConfigInvalidSubnetCIDRNotContainedWithinVirtualNetwork" Message="Subnet Prefix '10.241.0.0/16' specified for IngressApplicationGateway addon is not contained within the AKS Agent Pool's Virtual Network address prefixes '[10.224.0.0/12]'." Target="AddonProfiles.IngressApplicationGateway"
with azurerm_kubernetes_cluster.test,
on terraform_plugin_test.tf line 11, in resource "azurerm_kubernetes_cluster" "test":
11: resource "azurerm_kubernetes_cluster" "test" {
--- FAIL: TestAccKubernetesCluster_addonProfileIngressApplicationGateway_subnetCIDR (125.31s)
FAIL
```

```
------- Stdout: -------
=== RUN   TestAccDataSourceKubernetesCluster_addOnProfileIngressApplicationGatewaySubnetCIDR
=== PAUSE TestAccDataSourceKubernetesCluster_addOnProfileIngressApplicationGatewaySubnetCIDR
=== CONT  TestAccDataSourceKubernetesCluster_addOnProfileIngressApplicationGatewaySubnetCIDR
testcase.go:110: Step 1/1 error: Error running apply: exit status 1
Error: creating Cluster: (Managed Cluster Name "acctestaks220509101114912432" / Resource Group "acctestRG-aks-220509101114912432"): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="IngressAppGwAddonConfigInvalidSubnetCIDRNotContainedWithinVirtualNetwork" Message="Subnet Prefix '10.241.0.0/16' specified for IngressApplicationGateway addon is not contained within the AKS Agent Pool's Virtual Network address prefixes '[10.224.0.0/12]'." Target="AddonProfiles.IngressApplicationGateway"
with azurerm_kubernetes_cluster.test,
on terraform_plugin_test.tf line 12, in resource "azurerm_kubernetes_cluster" "test":
12: resource "azurerm_kubernetes_cluster" "test" {
--- FAIL: TestAccDataSourceKubernetesCluster_addOnProfileIngressApplicationGatewaySubnetCIDR (129.39s)
FAIL
```

**Comment by ms-henglu** [comment]
@stephybun - thanks for pointing it out, I've fixed the testcases.

**Review by katbyte — CHANGES_REQUESTED** [review]
As we are updating the SDK can we please try and fix all the test failures?

image

**Comment by ms-henglu** [comment]
Hi @katbyte , would you please take another look? I've fixed the most of the test cases.

**Comment by katbyte** [comment]
Still have some test failures, some of which first occurrence was this pr: 
image

**Comment by ms-henglu** [comment]
Hi @katbyte - in another build, is it good to go? Because some of failed cases are caused by preview feature not enabled, and others seem to be temporary failure?
image

**Comment by katbyte** [comment]
these failures are pretty consistent:
image

could we please fix them? as for features the docs should have the required powershell to enable them (and if not it should be added) - paste it here and i'll get them enabled on the test subscription

thanks

**Comment by ms-henglu** [comment]
hi @katbyte , would you please help register the following features?
```
Microsoft.ContainerService/AKS-AzureDefender
Microsoft.ContainerService/AKS-EnableDualStack
```

**Comment by ms-henglu** [comment]
@katbyte - thanks for the code review, but I'll close this PR because there're no required features pending on this api-version. I opened another PR fixing the failed tests: https://github.com/hashicorp/terraform-provider-azurerm/pull/17124

**Pull request closed without merging**